src/algebra/group/defs.lean

```diff
@@ -240,13 +374,49 @@ by rw [←one_mul c, ←hba, mul_assoc, hac, mul_one b]
 
 end monoid
 
+@[simp, to_additive nsmul_zero']
+lemma npow_zero {M : Type u} [monoid M] (x : M) :
+  npow 0 x = 1 :=
+by simp [monoid.npow_eq_rec, npow_rec]
+
+@[simp] lemma npow_one {M : Type u} [monoid M] (x : M) :
+  npow 1 x = x :=
+by simp [monoid.npow_eq_rec, npow_rec]
+
+@[simp] lemma nsmul_one' {M : Type u} [add_monoid M] (x : M) :
+  nsmul 1 x = x :=
+by simp [add_monoid.nsmul_eq_rec, nsmul_rec]
+
+attribute [to_additive nsmul_one'] npow_one
+
+lemma npow_add {M : Type u} [monoid M] (m n : ℕ) (x : M) :
+  npow (m + n) x = npow m x * npow n x :=
+begin
+  simp only [monoid.npow_eq_rec],
+  induction m with m ih,
+  { rw [nat.zero_add, npow_rec, one_mul], },
+  { rw [nat.succ_add, npow_rec, ih, npow_rec, ← mul_assoc] }
+end
+
+lemma nsmul_add' {M : Type u} [add_monoid M] (m n : ℕ) (x : M) :
+  nsmul (m + n) x = nsmul m x + nsmul n x :=
+begin
+  simp only [add_monoid.nsmul_eq_rec],
+  induction m with m ih,
+  { rw [nat.zero_add, nsmul_rec, zero_add] },
+  { rw [nat.succ_add, nsmul_rec, ih, nsmul_rec, ← add_assoc] }
+end
+
+attribute [to_additive nsmul_add'] npow_add
```

Comment:
Can't `to_additive` save you repeating the proof here?

Reply:
No, it doesn't work here because there is a one in the proof that `to_additive` tries to convert to zero, but it shouldn't because it is `1 : ℕ`.